### PR TITLE
Feature/registered process

### PR DESCRIPTION
**Pull request by marcelloceschia**
allow async to be a atom, so it is posible to send async messages to a registered process

### CODE REVIEW

**Comment by seriyps** [comment]
https://github.com/epgsql/epgsql/pull/99

File: src/epgsql_sock.erl [hunk 1]
```diff
@@ -426,11 +426,11 @@ notify(State = #state{queue = Q}, Notice) ->
     end,
     State.
 
-notify_async(State = #state{async = Pid}, Msg) ->
-    case is_pid(Pid) of
-        true  -> Pid ! {epgsql, self(), Msg};
-        false -> false
-    end,
+notify_async(State = #state{async = Pid}, Msg) when is_pid(Pid); is_atom(Pid) ->
+    Pid ! {epgsql, self(), Msg},
```
Comment (seriyps) [inline src/epgsql_sock.erl]: It even will try to send messages to `undefined` and fail of course (if there is no process registered under `undefined` atom)

Comment (marcelloceschia) [inline src/epgsql_sock.erl]: this issue was fixed in the latest commit

Comment (marcelloceschia) [inline src/epgsql_sock.erl]: @seriyps somthing I need to change?

**Comment by seriyps** [comment]
Implementation looks good. But there are 2 things extra:

* Unittests
* There is another pull-request which adds exactly the same functionality: #99 

We are waiting for @davidw approval.

**Comment by marcelloceschia** [comment]
@seriyps you are right, #99 does nearly the same, sorry for reimplementing this again. But checking this code, it has the same issue you mentioned to me - there is no check if the named process is registered

**Comment by seriyps** [comment]
@marcelloceschia there is no check, but there is a `try ... catch` (`atom() ! any()` will raise exception if no process registered). I decided that it will be optimal fast-path. Since passing atom with no process registered on it is more like error and shouldn't happen. I was thinking about adding some logging in 'catch' clause, but epgsql don't have any logging at all, so, not sure if I should add.

**Comment by seriyps** [comment]
#99 with the same functionality was merged to `devel` branch. Thank you anyway!

**Pull request closed without merging**